### DISCUSSION TITLE
Enforcing lower case table names

### DISCUSSION THREAD

**Discussion by ext-mar** [prisma/prisma#11759] [Q&A] [answered]
Hi there!

We have recently started a microservice project and we have elected to use Prisma against a MySql database hosted in AWS.
We are two developers right now and we're running into an issue because of the case sensitivity of different operating system, one dev on windows, one dev on mac and AWS is using a Unix based system.

We have done two migrations and are running into problems because the table names in the generated outputs are not consistent, and the different OS handle that differently.

Is there a way to enforce generated tables name  to always be in lowercase without having to use @@map()? Sometime down the line  some developer is going to forget to add it, and we're going to have inconsistencies in table names across environments which as we have experienced will cause migrations to fail.

These two migrations work on Windows but not on AWS because it cannot find the lowercase named table that is created on Windows. On Mac when running the migrate command it doesn't recognize the existing table and creates a migration for a table with CamelCase naming.

![image](https://user-images.githubusercontent.com/97104270/153586903-07b1031c-f651-4fae-991b-f359b8ed1f15.png)

![image](https://user-images.githubusercontent.com/97104270/153586458-308f9a9a-55fe-48fb-b9c9-542e0e716f76.png)

![image](https://user-images.githubusercontent.com/97104270/153586495-ba978f96-cafc-4a2a-a1a4-9b72ced812c6.png)

**Comment by ext-mar** [comment, marked as answer]
We have ended up running the database in a Docker container instead of having locally installed instances. Seems to make the local development environment closer to the one in AWS RDS.

**Comment by RobertSandiford** [comment]
You can set "lower_case_table_names = 2" in my.cnf / my.ini under [mysqld] to preserve table name casing on windows

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
